src/hooks/nabla/useTokenOutAmount.ts

```diff
@@ -0,0 +1,132 @@
+}
+
+export function useTokenOutAmount<FormFieldValues extends FieldValues>({
+  fromAmountString,
+  fromToken,
+  toToken,
+  maximumFromAmount,
+  form,
+  slippage,
+}: UseTokenOutAmountProps<FormFieldValues>): UseTokenOutAmountResult {
+  const { router } = useGetAppDataByTenant('nabla').data || {};
+  const { setError, clearErrors } = form;
+
+  const debouncedAmountString = useDebouncedValue(fromAmountString, 800);
+
+  let debouncedAmountBigDecimal: Big | undefined;
+  try {
+    debouncedAmountBigDecimal = debouncedAmountString !== undefined ? new Big(debouncedAmountString) : undefined;
+  } catch {
+    debouncedAmountBigDecimal = undefined;
+  }
+
+  const enabled =
+    router !== undefined &&
+    fromToken !== undefined &&
+    toToken !== undefined &&
+    debouncedAmountBigDecimal !== undefined &&
+    debouncedAmountBigDecimal?.gt(new Big(0)) &&
+    (maximumFromAmount === undefined || debouncedAmountBigDecimal.lte(maximumFromAmount));
+
+  const fromTokenDecimals = fromToken?.decimals;
+  const amountIn =
+    fromTokenDecimals !== undefined && debouncedAmountBigDecimal !== undefined
+      ? multiplyByPowerOfTen(debouncedAmountBigDecimal, fromTokenDecimals).round(0, 0).toString()
+      : undefined;
+
+  const { isLoading, fetchStatus, data, error, refetch } = useContractRead<TokenOutData | undefined>(
+    [cacheKeys.tokenOutAmount, fromToken?.id, toToken?.id, amountIn],
+    {
+      abi: routerAbi,
+      address: router,
+      method: 'getAmountOut',
+      args: [amountIn, [fromToken?.id, toToken?.id]],
+      noWalletAddressRequired: true,
+      queryOptions: {
+        ...activeOptions['30s'],
+        enabled,
+        onError: (err) => {
+          console.error(err);
+        },
+      },
+      parseSuccessOutput: (data) => {
+        if (toToken === undefined || fromToken === undefined || debouncedAmountBigDecimal === undefined)
+          return undefined;
+        const amountOut = parseContractBalanceResponse(toToken.decimals, data[0]);
+        const effectiveExchangeRate = stringifyBigWithSignificantDecimals(
+          amountOut.preciseBigDecimal.div(debouncedAmountBigDecimal),
+          6,
+        );
+        const minAmountOut = subtractBigDecimalPercentage(amountOut.preciseBigDecimal, slippage);
+
+        return {
+          amountOut,
+          effectiveExchangeRate,
+          swapFee: parseContractBalanceResponse(toToken.decimals, data[1]),
+          minAmountOut: stringifyBigWithSignificantDecimals(minAmountOut, 2),
+        };
+      },
+      parseError: (error) => {
+        switch (error.type) {
+          case 'error':
+            return 'Something went wrong';
+          case 'panic':
+            return error.errorCode === 0x11 ? 'The input amount is too large' : 'Something went wrong';
+          case 'reverted':
+            return error.description === 'SwapPool: EXCEEDS_MAX_COVERAGE_RATIO'
+              ? 'The input amount is too large'
+              : 'Something went wrong';
```

Comment:
Can we use an error message other than `'The input amount is too large'`? As a user, I don't know how it is too large, and I don't even know which amount this is referring to. It's not obvious that the error is about the coverage ratio of the pool.